.github/workflows/maven.yaml

```diff
@@ -3,33 +3,24 @@
 
 name: mvn verify
 
-on:
-  push:
-    branches: [ main ]
-  pull_request:
-    branches: [ main ]
+on: [push, pull_request, workflow_dispatch]
```

Comment:
Adding workflow_dispatch just allows triggering the build manually, but if GitHub Actions is set up properly, they will have already been run, and can be re-run manually. The dispatch isn't needed, unless you're adding extra config.

The reason it is restricted to main for pushes and pull requests is to prevent abuse of our Actions quota if there are multiple feature branches concurrently under development. Forks can modify their Actions configs however they want, but for this project, we don't want a bunch of actions running on all the branches if there's more than one created. If a branch is created, it can have a modified config file to support building on additional branches, if that is something we want to do.

Reply:
Typical reason I have workspace_dispatch is if I haven't built something in a long time and it was showing failed yet GH removed all the data. Probably rare case but that makes it a bit easier but if you prefer it gone, no worries as its less used item.

I don't think there are many branches on the upstreams here to pass any quota's to start with.  I have some pretty wild builds that don't surpass those and I'm not sure they really block OSS that much.  To see a bit of extreme, this repository is extreme https://github.com/psi-probe/psi-probe/actions.  I'm getting ready to split jakarta from javax over there and have bots running on both the namespace types on different branches and extensively run many jdks and many platforms.  Oshi is probably another out there that is pretty extensive in what it does.

The problem I face as down stream is that I tend to keep around a lot of branches that are work in progress stuff and its a pain to have to keep reworking GHA to work, then rebase it out just to raise a PR.  I presume that would bother others too or worse we get PRs where someone turns something on we don't want.  Think 'master' branch usage as its still the most used as far as I can tell and upstreams changing don't require downstreams to change at all.  Many won't.  So that was the real reason there to just relax.  No one can push stuff up to blow any quota on us anyways as far as I know.  New users to contribute don't even run unless we allow them. And those are only under the PR portion.  Outside users cannot add branches to the repo in other words.  Thus my obvious preference.  But I'm fine if maybe you could add 'main' and 'master' as acceptable.  Most often I'm toggling between those two and given any downstream will still be 'master' unless they make a change, that would make sense to have for 99% of use cases.